packages/core/src/common/performance/stopwatch.ts

```diff
@@ -50,7 +50,7 @@ export abstract class Stopwatch {
     @inject(ILogger)
     protected readonly logger: ILogger;
 
-    protected constructor(protected readonly defaultLogOptions: LogOptions) {
+    constructor(protected readonly defaultLogOptions: LogOptions) {
```

Comment:
It seems to make sense to remove the `protected` keyword but I didn't see where else it might have caused issues in this PR? I am not asking to get rid of this change, just trying to understand what I missed.

Reply:
`protected constructor` does some weird stuff to the constructor return type, which the `@injectable` annotation cannot deal with. See:

![image](https://user-images.githubusercontent.com/4377073/235442382-626c620c-59d9-4de5-a382-a22bb3c09edb.png)